GMX01 - Large Open/Increase Positions (#266)
NethermindEth/Forta-Agents: GMX-Bots/Large-Open-and-Increase-Positions/README.md

# Large Open/Increase Position Size Bot

## Description

> This bot detects when an open/increase position is large. The bot listens to `UpdatePosition` and `IncreasePosition` event emissions on GMX's Vault Contract and creates a finding if the position `size` exceeds the `threshold` 
> 

Reviewer: Suggesting an update for the following reasons:
1. first part is just for a better explanation. ( Feel free to improve it) 
2. We are not comparing the position size in both cases, so let's at least mention the correct parameter name used in comparison, ie `sizeDelta`. 

```suggestion
This bot detects a large size position opening and a large increase in an existing  position. It listens to `UpdatePosition` and `IncreasePosition` event emissions on GMX's Vault Contract and creates a finding if `sizeDelta` exceeds a `threshold`. 
```

> This bot detects when an open/increase position is large. The bot listens to `UpdatePosition` and `IncreasePosition` event emissions on GMX's Vault Contract and creates a finding if the initial position `size` exceeds the `threshold` 

Reviewer 2: ```suggestion
This bot detects when an open/increase position is large. The bot listens to `UpdatePosition` and `IncreasePosition` event emissions on GMX's Vault Contract and creates a finding if the position `size` exceeds the `threshold` 
```
Why to say `initial`?

Author: fixed

> The `threshold` can be adjusted in **src/network.ts**.

## Supported Chains

- Arbitrum
- Avalanche

## Alerts

- GMX-1-1

>   - Fired when the position `size` of the `UpdatePosition` event  which exceeds the `threshold` is equivalent to the `sizeDelta` of the `IncreasePosition` event

Reviewer 2: First, this description is a bit messy. You can say that `sizeDelta` exceeds the `threshold` because that is the comparison you are doing, as in this case it is equal to size, the sentence is "correct", but that is not the ideal explanation. 
Second, from my understanding what you are trying to detect here is an opening which I can confirm reading the finding object description/name, so it will be worthy mentioning here that is fired when a newly Opened position is large.

  - Severity is always set to "Info".
  - Type is always set to "Info".
  - Metadata contains:
    - `gmxVault`: Address of GMX Vault contract.
    - `account`: Address of position owner.
    - `positionSize`: Size of the opened position.
    - `positionKey`: Key of the opened position.

- GMX-1-2

>   - Fired only when the `size` of the `UpdatePosition` event which exceeds the `threshold` is greater than the `sizeDelta` of the `IncreasePosition` event

Reviewer 2: more or less the same happens here, you are not comparing the size

  - Severity is always set to "Info".
  - Type is always set to "Info".
  - Metadata contains:
    - `gmxVault`: Address of GMX Vault contract.
    - `account`: Address of position owner.
    - `initialPositionSize`: Initial position size before increment.
    - `positionIncrementSize`: Size of the increased position 
    - `finalPositionSize`: Final position size after increment
    - `positionKey`: Key of the increased position.

## Test Data

The bot behaviour can be verified with the following transactions:
- [0xa1497f703ad237a5f43a2df93fe961394e5df2ad548d4cb02c45a865d8fda591](https://arbiscan.io/tx/0xa1497f703ad237a5f43a2df93fe961394e5df2ad548d4cb02c45a865d8fda591) - 
`Arbitrum Mainnet` - UpdatePosition event 
  > Position size - 147986.41

- [0x70562d5cda648946dcefaf394f1db5591122b61a7924fc17e473fa0372a13213](https://arbiscan.io/tx/0x70562d5cda648946dcefaf394f1db5591122b61a7924fc17e473fa0372a13213) -
`Arbitrum Mainnet` - IncreasePosition event 

>   > Position increment size - 73142.32

Reviewer 2: That is not the Position increment size

Author: fixed that :)
